lib/uiautomator2.js

```diff
@@ -97,6 +97,38 @@ class UiAutomator2Server {
     return resultInfo;
   }
 
+  /**
+   * @typedef {Object} PackageInfo
+   * @property {string} installState
+   * @property {string} appPath
+   * @property {string} appId
+   */
+
+  /**
+   * If the given package information includes servers should uninstall befoer installing new UIA2 servers.
+   * For example, if the servers on the device is newer than servers UIA2 driver wants to use for the session.
+   *
+   * @param {PackageInfo[]} packagesInfo
+   * @returns {boolean}
+   */
+  shouldUninstallServerPackages(packagesInfo = []) {
+    return (packagesInfo.some(({installState}) => installState === this.adb.APP_INSTALL_STATE.NEWER_VERSION_INSTALLED)
+    && !packagesInfo.every(({installState}) => installState === this.adb.APP_INSTALL_STATE.NOT_INSTALLED));
+  }
+
+  /**
+   * If the given package information includes servers which can install with current UIA2 driver has.
```

Comment:
Checks if server components should be installed on the device under test in scope of the current driver session